### PR TITLE
Make E2E Kind node images reproducible

### PR DESCRIPTION
**Pull request by gergely-szabo-sap**
### Summary

Ensure E2E tests consistently use the configured Kind node image version.

### Changes

- Export `KIND_NODE_IMAGE_TAG` from the `Makefile`.
- Configure E2E cluster creation to use `kindest/node:<tag>`.
- Add a default image tag for direct `go test` invocations outside Make.

### Why

Reproducible E2E testing requires the Kubernetes node image to be explicit and consistent across local and CI environments. This prevents tests from silently using different default images, reducing environment-specific failures and making test results easier to compare and debug.

### Scope

This change only affects Kind node image selection for E2E tests; it does not change provider behavior or production resources.

### CODE REVIEW

File: test/e2e/main_test.go [hunk 1]
```diff
@@ -180,6 +189,16 @@ func SetupClusterWithCrossplane(namespace string) {
 	)
 }
 
+// resolveKindNodeImageTag returns the configured Kind node image tag. Keep a
+// default here for direct `go test -tags=e2e` invocations that do not go
+// through the Makefile.
```
Comment (n-losse) [inline test/e2e/main_test.go:194]: is that something we want to encourage? ideally i'd imagine all e2e tests using the Makefile

Comment (gergely-szabo-sap) [reply test/e2e/main_test.go:194]: I would not encourage it, but we cannot prevent it. This is just a precaution to make this work even in such edge cases.

Comment (n-losse) [reply test/e2e/main_test.go:194]: making it fail would be a way of preventing it, kind of. :D 

if we want to make it work, can we make that use the latest instead of some fixed version hidden in here?

Comment (gergely-szabo-sap) [reply test/e2e/main_test.go:194]: The question is: which is worse? Hiding the node version or making it irreproducible? I lean towards hiding the node version as better. But I can be convinced :)

Comment (n-losse) [reply test/e2e/main_test.go:194]: my argument would be that the proper, reproducible way is to use the Makefile, and if we're enabling the test run without the Makefile as a side path, then that using the latest image is less surprising to the user than it using some obscure hardcoded version

Comment (gergely-szabo-sap) [reply test/e2e/main_test.go:194]: I agree to disagree. In my view, an irreproducible test can be more surprising than one that uses a hardcoded version. An irreproducible test is more difficult to debug and fix. An irreproducible test can generate obscure issue reports: "It was working on my machine but not on yours". Irreproducible tests are the root of all evil, the NULL pointer of testing :)

Comment (gergely-szabo-sap) [reply test/e2e/main_test.go:194]: @n-losse, I added https://github.com/SAP/crossplane-provider-btp/pull/964/commits/8e766c73e98c0e07775ee11c15d85429e9a6ad94

This commit introduces a renovate rule to keep the hardcoded node versions in sync and up to date.

Comment (n-losse) [reply test/e2e/main_test.go:194]: i think my preferences in order would be 

- make runs without makefile fail 
- make them use latest
- make them use fixed version 

but I'll be able to live with it :D

**Review by n-losse — APPROVED** [review]
